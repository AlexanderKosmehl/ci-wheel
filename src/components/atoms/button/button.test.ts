import {
  describe, expect, it, vi,
} from 'vitest';
import generateButton from './button';

describe('generateButton', () => {
  const onClick = vi.fn(() => {});

  const button = generateButton({
    content: 'Text',
    classes: ['class', 'class2'],
    onClick,
  });

  it('generates a button component', () => {
    expect(button).toMatchSnapshot();
  });

  it('adds working clickListener', () => {
    button.click();
    expect(onClick).toBeCalled();
  });
});
