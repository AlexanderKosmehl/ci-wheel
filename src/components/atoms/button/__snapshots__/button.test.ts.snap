// Vitest Snapshot v1

exports[`generateButton > generates a button component 1`] = `
<button
  class="class class2"
>
  Text
</button>
`;
